docs/content/guides/dagster-pipes/kubernetes.mdx

```diff
@@ -0,0 +1,243 @@
+      kube_context="<YOUR_CONTEXT_NAME>"
+  ).get_materialize_result()
+```
+
+Here’s what we did in this example:
+
+- Created an asset named `k8s_pipes_asset`
+
+- Provided <PyObject object="AssetExecutionContext" /> as the `context` argument to the asset. This object provides access to system APIs such as resources, config, and logging.
+
+- Specified a resource for the asset to use, <PyObject module="dagster_k8s" object="PipesK8sClient" />, which is a pre-built Dagster resource that allows you to quickly get Pipes working with your Kubernetes workspace.
+
+  We also specified the following for the resource:
+
+  - `context` - The asset's `context` (<PyObject object="AssetExecutionContext" />) data
+  - `image` - The Kubernetes image we want to use
+  - `kubeconfig_file` - A kube-config file containing authentication information to pass to `kubernetes.config.load_kube_config`
+  - `kube_context` - A context argument to pass to `kubernetes.config.load_kube_config`
+
+- Passed the `context`, `image`, `kubeconfig_file`, and `kube_context` to the `run` method of <PyObject module="dagster_k8s" object="PipesK8sClient" />. This method submits the provided cluster information to the Kubernetes API and then runs the specified `image`.
+
+- Returned a <PyObject object="MaterializeResult" /> object representing the result of execution. This is obtained by calling `get_materialize_result` on the <PyObject module="dagster_pipes" object="PipesClientCompletedInvocation" /> object returned by `run` after the execution in Kubernetes has completed. **Note**: Execution can take several minutes even for trivial scripts due to cluster provisioning times.
+
+### Step 1.2: Create Dagster Definitions
+
+Next, you’ll add the asset and Kubernetes resource to your project’s code location via the <PyObject object="Definitions" /> object. This makes the resource available to other Dagster definitions in the project.
+
+Copy and paste the following to the bottom of `dagster_k8s_pipes.py`:
+
+```python
+# dagster_k8s_pipes.py
+
+defs = Definitions(
+  assets=[k8s_pipes_asset],
+  resources={
+    "k8s_pipes_client": PipesK8sClient(),
+  },
+)
+```
+
+At this point, `dagster_k8s_pipes.py` should look like the following:
+
+```python
+# dagster_k8s_pipes.py
+
+from dagster import AssetExecutionContext, Definitions, asset
+from dagster_k8s import PipesK8sClient
+
+
+@asset
+def k8s_pipes_asset(context: AssetExecutionContext, k8s_pipes_client: PipesK8sClient):
+  return k8s_pipes_client.run(
+      context=context,
+      image="pipes-materialize:latest",
+      kubeconfig_file=os.path.expanduser("~/.kube/config"),
+      kube_context="<YOUR_CONTEXT_NAME>"
+  ).get_materialize_result()
+
+
+defs = Definitions(
+  assets=[k8s_pipes_asset],
+  resources={
+    "k8s_pipes_client": PipesK8sClient(),
+  },
+)
+```
+
+---
+
+## Step 2: Write a script for execution on Kubernetes
+
+The next step is to write the code that will be executed on Kubernetes.
+
+TODO
```

Comment:
For Databricks, we have this as the intro to this section:

> The next step is to write the code that will be executed on Databricks. In the Databricks task specification in Step 1, we referenced a file `dbfs:/my_python_script.py` in the `spark_python_task`. We'll create this script from scratch and upload it to DBFS. You can use the Databricks UI or run a command from a shell to do this.

What's the equivalent here for Kubernetes?

Reply:
[2] we will need to package up this code in to a container image by using a `Dockerfile` . That image is what is referenced in [1] so we may want to re-order steps